crates/sui-types/src/test_checkpoint_data_builder.rs

```diff
@@ -0,0 +1,826 @@
+
+        // Verify object appears in output objects
+        assert!(tx
+            .output_objects
+            .iter()
+            .any(|obj| obj.id() == created_obj_id));
+
+        // Verify effects show object creation
+        assert!(tx
+            .effects
+            .created()
+            .iter()
+            .any(|((id, ..), owner)| *id == created_obj_id
+                && owner.get_owner_address().unwrap()
+                    == TestCheckpointDataBuilder::derive_address(0)));
+    }
+
+    #[test]
+    fn test_object_mutation() {
+        let checkpoint = TestCheckpointDataBuilder::new(1)
+            .start_transaction(0)
+            .create_owned_object(0)
+            .finish_transaction()
+            .start_transaction(0)
+            .mutate_object(0)
+            .finish_transaction()
+            .build_checkpoint();
+
+        let tx = &checkpoint.transactions[1];
+        let obj_id = TestCheckpointDataBuilder::derive_object_id(0);
+
+        // Verify object appears in input and output objects
+        assert!(tx.input_objects.iter().any(|obj| obj.id() == obj_id));
+        assert!(tx.output_objects.iter().any(|obj| obj.id() == obj_id));
+
+        // Verify effects show object mutation
+        assert!(tx
+            .effects
+            .mutated()
+            .iter()
+            .any(|((id, ..), _)| *id == obj_id));
+    }
+
+    #[test]
+    fn test_object_deletion() {
+        let checkpoint = TestCheckpointDataBuilder::new(1)
+            .start_transaction(0)
+            .create_owned_object(0)
+            .finish_transaction()
+            .start_transaction(0)
+            .delete_object(0)
+            .finish_transaction()
+            .build_checkpoint();
+
+        let tx = &checkpoint.transactions[1];
+        let obj_id = TestCheckpointDataBuilder::derive_object_id(0);
+
+        // Verify object appears in input objects but not output
+        assert!(tx.input_objects.iter().any(|obj| obj.id() == obj_id));
+        assert!(!tx.output_objects.iter().any(|obj| obj.id() == obj_id));
+
+        // Verify effects show object deletion
+        assert!(tx.effects.deleted().iter().any(|(id, ..)| *id == obj_id));
+    }
+
+    #[test]
+    fn test_object_wrapping() {
+        let checkpoint = TestCheckpointDataBuilder::new(1)
+            .start_transaction(0)
+            .create_owned_object(0)
+            .finish_transaction()
+            .start_transaction(0)
+            .wrap_object(0)
+            .finish_transaction()
+            .start_transaction(0)
+            .unwrap_object(0)
+            .finish_transaction()
+            .build_checkpoint();
+
+        let tx = &checkpoint.transactions[1];
+        let obj_id = TestCheckpointDataBuilder::derive_object_id(0);
+
+        assert!(tx.input_objects.iter().any(|obj| obj.id() == obj_id));
+        assert!(!tx.output_objects.iter().any(|obj| obj.id() == obj_id));
+        assert!(tx.effects.wrapped().iter().any(|(id, ..)| *id == obj_id));
+
```

Comment:
```suggestion
// Verify unwrapped object does not appear in input objects, but appears in output objects and effects
```

I found the comments for the other test cases to be useful, and not redundant, so might be worth adding to the rest?